src/hypertable.c

```diff
@@ -1712,11 +1686,109 @@ ts_validate_replication_factor(int32 replication_factor, bool is_null, bool is_d
 				 errhint("A hypertable's replication factor must be between 1 and %d.",
 						 PG_INT16_MAX)));
 
+	return (int16)(replication_factor & 0xFFFF);
+}
+
+static int16
+hypertable_validate_create_call(const char *hypertable_name, bool distributed,
+								bool distributed_is_null, int32 replication_factor,
+								bool replication_factor_is_null, ArrayType *data_node_arr,
+								List **data_nodes)
+{
+	bool distributed_local_error = false;
+
+	if (!distributed_is_null && !replication_factor_is_null)
+	{
+		/* create_hypertable(distributed, replication_factor) */
+		if (!distributed)
+			distributed_local_error = true;
+	}
+	else if (!distributed_is_null)
+	{
+		/* create_hypertable(distributed) */
+		switch (ts_guc_hypertable_distributed_default)
+		{
+			case HYPERTABLE_DIST_AUTO:
+			case HYPERTABLE_DIST_DISTRIBUTED:
+				if (distributed)
+					replication_factor = ts_guc_hypertable_replication_factor_default;
+				else
+				{
+					/* creating regular hypertable according to the policy */
+				}
+				break;
+			case HYPERTABLE_DIST_LOCAL:
+				if (distributed)
+					replication_factor = ts_guc_hypertable_replication_factor_default;
+				else
+					distributed_local_error = true;
+				break;
+		}
+	}
+	else if (!replication_factor_is_null)
+	{
+		/* create_hypertable(replication_factor) */
+		switch (ts_guc_hypertable_distributed_default)
+		{
+			case HYPERTABLE_DIST_AUTO:
+			case HYPERTABLE_DIST_DISTRIBUTED:
+				/* distributed and distributed member */
+				distributed = true;
+				break;
+			case HYPERTABLE_DIST_LOCAL:
+				distributed_local_error = true;
+				break;
+		}
+	}
+	else
+	{
+		/* create_hypertable() */
+		switch (ts_guc_hypertable_distributed_default)
+		{
+			case HYPERTABLE_DIST_AUTO:
+				distributed = false;
+				break;
+			case HYPERTABLE_DIST_LOCAL:
+				distributed_local_error = true;
+				break;
+			case HYPERTABLE_DIST_DISTRIBUTED:
+				replication_factor = ts_guc_hypertable_replication_factor_default;
+				distributed = true;
+				break;
+		}
+	}
+
+	if (distributed_local_error)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("operation not supported"),
+				 errdetail("Local hypertables cannot set replication_factor.")));
```

Comment:
Just use the main message, it is more informative.
```suggestion
				 errmsg("local hypertables cannot set a replication_factor")));
```